pycbc/types/optparse.py

```diff
@@ -214,6 +215,64 @@ def __call__(self, parser, namespace, values, option_string=None):
                 raise ValueError(err_msg)
         setattr(namespace, self.dest, items)
 
+class ExtraArgsOptionAction(argparse.Action):
+    # Initialise the same as the standard 'append' action
+    def __init__(self,
+                 option_strings,
+                 dest,
+                 nargs='+',
+                 const=None,
+                 default=None,
+                 type=None,
+                 choices=None,
+                 required=False,
+                 help=None,
+                 metavar=None):
+        if type is not None:
+            self.internal_type = type
+        else:
+            self.internal_type = str
+        new_default = None
+        if nargs == 0:
+            raise ValueError('nargs for append actions must be > 0; if arg '
+                             'strings are not supplying the value to append, '
+                             'the append const action may be more appropriate')
+        if const is not None and nargs != argparse.OPTIONAL:
+            raise ValueError('nargs must be %r to supply const'
+                             % argparse.OPTIONAL)
+        super(ExtraArgsOptionAction, self).__init__(
+            option_strings=option_strings,
+            dest=dest,
+            nargs=nargs,
+            const=const,
+            default=new_default,
+            type=str,
+            choices=choices,
+            required=required,
+            help=help,
+            metavar=metavar)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        # Again this is modified from the standard argparse 'append' action
+        err_msg = "Issue with option: %s \n" %(self.dest,)
+        err_msg += "Received value: %s \n" %(' '.join(values),)
+        if getattr(namespace, self.dest, None) is None:
+            setattr(namespace, self.dest, DictWithDefaultReturn())
```

Comment:
Do we want to use DictWithDefaultReturn here? or just populate a regular dictionary?